boxsdk/util/api_call_decorator.py

```diff
@@ -0,0 +1,58 @@
+# coding: utf-8
+
+from __future__ import unicode_literals, absolute_import
+
+from functools import update_wrapper, wraps
+
+
+def api_call(method):
+    """
+    Designates the decorated method as one that makes a Box API call.
+    The decorated method can then accept a new keyword argument `extra_network_parameters`,
+    a dictionary of key-value pairs to be passed to the network layer for API
+    calls made by the method.
+
+    The decorated method must belong to a subclass of `Cloneable` as using this
+    decorator and then passing a `extra_network_parameters` parameter to the method will cause
+    the object's clone method to be called.
+
+    :param method:
+        The method to decorate.
+    :type method:
+        `callable`
+    :return:
+        A wrapped method that can pass extra request data to the network layer.
+    :rtype:
+        :class:`APICallWrapper`
+    """
+    return APICallWrapper(method)
+
+
+class APICallWrapper(object):
+
+    def __init__(self, func_that_makes_an_api_call):
+        super(APICallWrapper, self).__init__()
+        self._func_that_makes_an_api_call = func_that_makes_an_api_call
+        update_wrapper(self, func_that_makes_an_api_call)
+
+    def __get__(self, _instance, owner):
+        @wraps(self._func_that_makes_an_api_call)
```

Comment:
You added `@wraps` which fixes the name/docstring problem, but just re-copying this here as a note to my future self, in case we need to improve the performance of this code later on:

This will cause us to need to construct a new function every time we want to make an API call.

To be more economical, we could try something like this:

```python
class APICallWrapper(object):
    def __init__(self, func_that_makes_an_api_call, instance=None, owner=None):
        ...
        self._instance = instance
        self._owner = owner

    def __get__(self, instance, owner):
        return self.__class__(func_that_makes_an_api_call, instance=instance, owner=owner)

    def __call__(*args, **kwargs):
        ...
```

As another idea, we can consider using https://pypi.python.org/pypi/wrapt to power the decorator, so that we don't have to think at all about the descriptor. Though that would introduce its own slowness.